types/recurly__recurly-js/lib/google-pay/native.d.ts

```diff
@@ -0,0 +1,336 @@
+   */
+  checkoutOption?:
+    /**
+     * Standard text applies for the given totalPriceStatus (default).
+     */
+    | 'DEFAULT'
+    /**
+     * The selected payment method is charged immediately after the payer confirms their selections.
+     * This option is only available when totalPriceStatus is set to FINAL.
+     */
+    | 'COMPLETE_IMMEDIATE_PURCHASE';
+};
+
+/**
+ * Provides information about any Offers currently applied to a transaction.
+ */
+export type GooglePayOfferDetail = {
+  /**
+   * The promotional code associated with the Offer. Should be the same as the code entered into the payment sheet.
+   */
+  redemptionCode: string;
+
+  /**
+   * A description of the promotion applied by the Offer code.
+   */
+  description: string;
+};
+
+/**
+ * Provides information about any Offers currently applied to a transaction.
+ */
+export type GooglePayOfferInfo = {
+  offers: GooglePayOfferDetail[];
+};
+
+/**
+ * Set shipping restrictions.
+ */
+export type GooglePayShippingAddressParameters = {
+   /**
+    * ISO 3166-1 alpha-2 country code values of the countries where shipping is allowed.
+    * If this object isn't specified, all shipping address countries are allowed.
+    */
+  allowedCountryCodes?: string[];
+
+  /**
+   * Set to true if a phone number is required for the provided shipping address.
+   */
+  phoneNumberRequired?: boolean;
+};
+
+/**
+ * Shipping option selection
+ */
+export type GooglePaySelectionOption = {
+  /**
+   * The developer can put any value that needs to be returned in PaymentData.
+   */
+  id: string;
+
+  /**
+   * The label to be displayed as the option.
+   */
+  label: string;
+
+  /**
+   * A descriptive text that is displayed below the option label.
+   */
+  description?: string;
+};
+
+/**
+ * Set shipping options.
+ */
+export type GooglePayShippingOptionParameters = {
+  /**
+   * All of the shipping options available for the current request.
+   */
+  shippingOptions: GooglePaySelectionOption[];
+
+  /**
+   * An identifier to the default selected shipping option.
+   * If this field isn't provided, the first option is the default option.
+   */
+  defaultSelectedOptionId?: string;
+};
+
+/**
+ * Configuration object for Google Pay API.
+ */
+export type GooglePayPaymentDataRequest = {
+  /**
+   * Information about the merchant that requests payment data.
+   */
+  merchantInfo?: GooglePayMerchantInfo;
+
+  /**
+   * Details about the authorization of the transaction based upon whether the user agrees to the transaction or not.
+   */
+  transactionInfo?: GooglePayTransactionInfo;
+
+  /**
+   * Specifies which offers to apply when the payment sheet first loads.
+   */
+  offerInfo?: GooglePayOfferInfo;
+
+  /**
+   * Request an email address.
+   */
+  emailRequired?: boolean;
+
+  /**
+   * Request an shipping address.
+   */
+  shippingAddressRequired?: boolean;
+
+  /**
+   * 	If shippingAddressRequired is set to true, specify shipping address restrictions.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/recurly__recurly-js/lib/google-pay/native.d.ts;
  Line: 216;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.